### PR TITLE
Add ability to see message headers from Vantiq Camel Component.

### LINKED ISSUE #394: camelComponent: add structured message option
There are times when camel components (not Vantiq) use the messages headers to obtain or provide information.  

Add a URL option to instruct the Vantiq camel component to provide the message (to or from Vantiq) in format matching Camel's structure.  Something like

```js
{
    header: { ... },
    message: { ... }
}
```

where header & message will be another Json object.  We will define a schema type for this structure.

This doesn't give low-level details (for some components, the `message` part may be well-structured, but it gives us some control and allows us to provide the headers as desired.  Since the Vantiq camel component did not do things this way from the start, this is an option -- use of the option allows for backward compatibility.

### PR DESCRIPTION
**Pull request by fhcarter**
Fixes #394 

Adds endpoint config flag (URL param) to ask for "structured' messages.  These are messages have the camel message  & header present, allowing caller to set headers & consumers to see them.

Added tests for same

Some other code cleanup as well.

### CODE REVIEW

File: camelComponent/src/main/java/io/vantiq/extsrc/camel/VantiqConsumer.java [hunk 1]
```diff
@@ -60,31 +64,30 @@ protected void doStop() throws Exception {
...
      * A handler for dealing with publishes to Camel.
      */
-    private final Handler<ExtensionServiceMessage> publishHandler = new Handler<ExtensionServiceMessage>() {
+    private final Handler<ExtensionServiceMessage> publishHandler = new Handler<>() {
         @Override
         public void handleMessage(ExtensionServiceMessage message) {
             // When we get a message, process it in the background...
-            executorService.submit( () -> {
-                processMessage(message, InOnly);
-            });
```
Comment (fhcarter) [inline camelComponent/src/main/java/io/vantiq/extsrc/camel/VantiqConsumer.java:74]: just code cleanup...

File: camelComponent/src/main/java/io/vantiq/extsrc/camel/VantiqConsumer.java [hunk 2]
```diff
@@ -96,7 +99,25 @@ private void processMessage(Object msg, ExchangePattern pattern) {
         } else {
             ExtensionServiceMessage message = (ExtensionServiceMessage) msg;
             Object msgBody = message.getObject();
+            Map<String, Object> camelHdrs = null;
+            Object camelBody = null;
```
Comment (fhcarter) [inline camelComponent/src/main/java/io/vantiq/extsrc/camel/VantiqConsumer.java:103]: below is the meat of the change

File: camelComponent/src/main/java/io/vantiq/extsrc/camel/VantiqProducer.java [hunk 3]
```diff
@@ -98,6 +99,17 @@ public void process(Exchange exchange) throws Exception {
         if (ra instanceof String) {
             responseAddr = (String) ra;
         }
```
Comment (fhcarter) [inline camelComponent/src/main/java/io/vantiq/extsrc/camel/VantiqProducer.java:101]: and here...

File: camelComponent/src/test/java/io/vantiq/extsrc/camel/VantiqComponentTest.java [hunk 4]
```diff
@@ -109,18 +129,9 @@ public void testVantiqProducerJson() throws Exception {
             
             assert item != null;
             sendBody(routeStartUri, item);
+            //noinspection rawtypes
             Map lastMsg = fc.getLastMessageAsMap();
```
Comment (fhcarter) [inline camelComponent/src/test/java/io/vantiq/extsrc/camel/VantiqComponentTest.java:133]: code moved to shared method since it's used a lot

**Comment by fhcarter** [comment]
[Jenkins run](https://jenkins.vantiq.com/jenkins/job/vantiq-extension-sources-branch/141/)

**Review by jcguion — APPROVED** [review]
LGTM

**Pull request merged**